src/v0/destinations/marketo_bulk_upload/util.js

```diff
@@ -1,98 +1,343 @@
       destType: 'marketo_bulk_upload',
       feature: 'transformation',
-    },
-  );
-  const ACCESS_TOKEN_FETCH_ERR_MSG = 'Error during fetching access token';
-  if (resp.success) {
-    if (resp.response && resp.response.data && resp.response.data.access_token) {
-      return resp.response.data.access_token;
+    });
+
+    // sample response : {response: '[ENOTFOUND] :: DNS lookup failed', status: 400}
+    if (!isHttpStatusSuccess(resp.status)) {
+      throw new NetworkError(
+        'Could not retrieve authorisation token',
+        resp.status,
+        {
+          [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(resp.status),
+        },
+        resp,
+      );
     }
-    const status = resp?.response?.status || 400;
-    throw new NetworkError(
-      'Could not retrieve authorisation token',
-      status,
-      {
-        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(status),
-      },
-      resp,
-    );
-  }
-  if (resp.response) {
-    // handle for abortable codes
-    if (
-      ABORTABLE_CODES.includes(resp.response.code) ||
-      (resp.response.code >= 400 && resp.response.code <= 499)
-    ) {
-      throw new AbortedError(resp.response.code, 400, resp);
-    } // handle for retryable codes
-    else if (RETRYABLE_CODES.includes(resp.response.code)) {
-      throw new RetryableError(resp.response.code, 500, resp);
-    } // handle for abortable codes
-    else if (resp.response.response) {
-      if (ABORTABLE_CODES.includes(resp.response.response.status)) {
-        throw new AbortedError(
-          resp.response.response.statusText || ACCESS_TOKEN_FETCH_ERR_MSG,
-          400,
-          resp,
-        );
-      } // handle for throttled codes
-      else if (THROTTLED_CODES.includes(resp.response.response.status)) {
-        throw new ThrottledError(
-          resp.response.response.statusText || ACCESS_TOKEN_FETCH_ERR_MSG,
-          resp,
+    if (resp.response?.success === false) {
+      handleCommonErrorResponse(resp, ACCESS_TOKEN_FETCH_ERR_MSG, FETCH_ACCESS_TOKEN, config);
+    }
+
+    // when access token is present
+    if (resp.response.access_token) {
+      /* This scenario will handle the case when we get the foloowing response
+      status: 200  
+      respnse: {"access_token":"<dummy-access-token>","token_type":"bearer","expires_in":0,"scope":"dummy@scope.com"}
+      wherein "expires_in":0 denotes that we should refresh the accessToken but its not expired yet. 
+      */
+      if (resp.response?.expires_in === 0) {
+        throw new RetryableError(
+          `Request Failed for marketo_bulk_upload, Access Token Expired (Retryable).`,
+          500,
         );
       }
-      // Assuming none we should retry the remaining errors
-      throw new RetryableError(
-        resp.response.response.statusText || ACCESS_TOKEN_FETCH_ERR_MSG,
-        500,
+      return resp.response.access_token;
+    }
+    return null;
```

Comment:
Whats the behaviour of returning `null` in cache call back do we store null for the given key? If thats true we need to throw error here and not check null outside